File: poscloud-angular/src/shared/entity/basic-data/retailer.ts
export class Retailer implements IRetailer {
    id: string;
    name: string;
    code: string;
    licenseKey: string;
    businessAddress: string;
    archivalLevel: string;
    orderCycle: string;
    telephone: string;
    isAction: boolean;
    branchCompany: string;
    department: string;
    slsmanId: string;
    slsmanName: string;
    area: string;
    orderMode: number;
    terminalType: number;
    businessType: string;
    authorizationCode: string;
    actionText: string;
    actionType: string;
    retailerAuthorizationCode: string;

    isDeleted: boolean;
    creationTime: Date;
    creatorUserId: number;
    lastModificationTime: Date;
    lastModifierUserId: number;
    deletionTime: Date;
    deleterUserId: number;
    constructor(data?: IRetailer) {
        if (data) {
            for (var property in data) {
                if (data.hasOwnProperty(property))
                    (<any>this)[property] = (<any>data)[property];
            }
        }
    }

    init(data?: any) {
        if (data) {
            this.id = data["id"];
            this.name = data["name"];
            this.code = data["code"];
            this.licenseKey = data["licenseKey"];
            this.businessAddress = data["businessAddress"];
            this.archivalLevel = data["archivalLevel"];
            this.orderCycle = data["orderCycle"];
            this.telephone = data["telephone"];
            this.isAction = data["isAction"];
            this.branchCompany = data["branchCompany"];
            this.department = data["department"];
            this.slsmanId = data["slsmanId"];
            this.slsmanName = data["slsmanName"];
            this.area = data["area"];
            this.orderMode = data["orderMode"];
            this.terminalType = data["terminalType"];
            this.businessType = data["businessType"];
            this.authorizationCode = data["authorizationCode"];
            this.retailerAuthorizationCode = data["retailerAuthorizationCode"];

            this.isDeleted = data["isDeleted"];
            this.creationTime = data["creationTime"];
            this.creatorUserId = data["creatorUserId"];
            this.lastModificationTime = data["lastModificationTime"];
            this.lastModifierUserId = data["lastModifierUserId"];
            this.deletionTime = data["deletionTime"];
            this.deleterUserId = data["deleterUserId"];
        }
    }

    static fromJS(data: any): Retailer {
        let result = new Retailer();
        result.init(data);
        return result;
    }

    static fromJSArray(dataArray: any[]): Retailer[] {
        let array = [];
        dataArray.forEach(result => {
            let item = new Retailer();
            item.init(result);
            array.push(item);
        });

        return array;
    }

    toJSON(data?: any) {
        data = typeof data === 'object' ? data : {};
        data["id"] = this.id;
        data["name"] = this.name;
        data["code"] = this.code;
        data["licenseKey"] = this.licenseKey;
        data["businessAddress"] = this.businessAddress;
        data["archivalLevel"] = this.archivalLevel;
        data["orderCycle"] = this.orderCycle;
        data["telephone"] = this.telephone;
        data["isAction"] = this.isAction;
        data["branchCompany"] = this.branchCompany;
        data["department"] = this.department;
        data["slsmanId"] = this.slsmanId;
        data["slsmanName"] = this.slsmanName;
        data["area"] = this.area;
        data["orderMode"] = this.orderMode;
        data["terminalType"] = this.terminalType;
        data["businessType"] = this.businessType;
        data["authorizationCode"] = this.authorizationCode;

        data["isDeleted"] = this.isDeleted;
        data["creationTime"] = this.creationTime;
        data["creatorUserId"] = this.creatorUserId;
        data["lastModificationTime"] = this.lastModificationTime;
        data["lastModifierUserId"] = this.lastModifierUserId;
        data["deletionTime"] = this.deletionTime;
        data["deleterUserId"] = this.deleterUserId;
        return data;
    }

    clone() {
        const json = this.toJSON();
        let result = new Retailer();
        result.init(json);
        return result;
    }
}
export interface IRetailer {
    id: string;
    name: string;
    code: string;
    licenseKey: string;
    businessAddress: string;
    archivalLevel: string;
    orderCycle: string;
    telephone: string;
    isAction: boolean;
    branchCompany: string;
    department: string;
    slsmanId: string;
    slsmanName: string;
    area: string;
    orderMode: number;
    terminalType: number;
    businessType: string;
    authorizationCode: string;
    actionText: string;
    actionType: string;
    retailerAuthorizationCode: string;

    isDeleted: boolean;
    creationTime: Date;
    creatorUserId: number;
    lastModificationTime: Date;
    lastModifierUserId: number;
    deletionTime: Date;
    deleterUserId: number;
}

File: poscloud-aspnet-core/src/HC.POSCloud.Core/Retailer/Retailer.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Abp.Domain.Entities.Auditing;
using HC.POSCloud.PosEnmus;
namespace HC.POSCloud.Retailers
{
    /// <summary>
    /// 零售客户
    /// </summary>
    [Table("Retailers")]
    public class Retailer : FullAuditedEntity<Guid>
    {

        /// <summary>
        /// 客户编码
        /// </summary>
        [Required]
        [StringLength(50)]
        public virtual string Code { get; set; }

        /// <summary>
        /// 客户姓名
        /// </summary>
        [Required]
        [StringLength(50)]
        public virtual string Name { get; set; }

        /// <summary>
        /// 许可证号
        /// </summary>
        [Required]
        [StringLength(50)]
        public virtual string LicenseKey { get; set; }

        /// <summary>
        /// 经营地址
        /// </summary>
        [StringLength(50)]
        public virtual string BusinessAddress { get; set; }

        /// <summary>
        /// 客户分档
        /// </summary>
        [StringLength(100)]
        public virtual string ArchivalLevel { get; set; }

        /// <summary>
        /// 订货周期
        /// </summary>
        [StringLength(100)]
        public virtual string OrderCycle { get; set; }

        /// <summary>
        /// 订货电话
        /// </summary>
        [StringLength(100)]
        public virtual string Telephone { get; set; }

        /// <summary>
        /// 客户状态
        /// </summary>
        [Required]
        public virtual bool IsAction { get; set; }

        /// <summary>
        /// 分公司 快照
        /// </summary>
        [StringLength(200)]
        public virtual string BranchCompany { get; set; }

        /// <summary>
        /// 市场部﻿ 快照
        /// </summary>
        [StringLength(100)]
        public virtual string Department { get; set; }

        /// <summary>
        /// 营销经理ID
        /// </summary>
        [StringLength(50)]
        public virtual string SlsmanId { get; set; }

        /// <summary>
        /// 营销经理名称
        /// </summary>
        [StringLength(50)]
        public virtual string SlsmanName { get; set; }

        /// <summary>
        /// 所在区域
        /// </summary>
        [StringLength(100)]
        public virtual string Area { get; set; }

        /// <summary>
        /// 订货方式(枚举 无、网上订货、电话订货、手机)
        /// </summary>
        public virtual int? OrderMode { get; set; }

        /// <summary>
        /// 终端类型(枚举 无、建议终端、普通终端、现代终端)
        /// </summary>
        public virtual int? TerminalType { get; set; }

        /// <summary>
        /// 商圈类型
        /// </summary>
        [StringLength(100)]
        public virtual string BusinessType { get; set; }

        /// <summary>
        /// 终端授权码
        /// </summary>
        [StringLength(100)]
        public virtual string AuthorizationCode { get; set; }
    }
}
